flexmeasures/data/models/reporting/tests/test_tibber_reporter.py

```diff
@@ -0,0 +1,252 @@
+        beliefs.append(
+            TimedBelief(
+                event_start=dt,
+                belief_horizon=timedelta(hours=24),
+                event_value=value,
+                sensor=sensor,
+                source=source,
+            )
+        )
+
+    return beliefs
+
+
+@pytest.fixture()
+def tibber_test_data(fresh_db, app):
+    db = fresh_db
+
+    tax = GenericAssetType(name="Tax")
+    price = GenericAssetType(name="Price")
+
+    db.session.add_all([tax, price])
+
+    # Belastingdienst
+
+    electricity_price = GenericAsset(name="Electricity Price", generic_asset_type=price)
+
+    VAT = GenericAsset(name="VAT", generic_asset_type=tax)
+
+    electricity_tax = GenericAsset(name="Energy Tax", generic_asset_type=tax)
+
+    db.session.add_all([electricity_price, VAT, electricity_tax])
+
+    # Taxes
+    BWV = Sensor("BWV", generic_asset=VAT, event_resolution=timedelta(days=365))
+    EB = Sensor(
+        "EB", generic_asset=electricity_tax, event_resolution=timedelta(days=365)
+    )
+
+    # Tibber Tariff
+    tibber_tariff = Sensor(
+        "Tibber Tariff",
+        generic_asset=electricity_price,
+        event_resolution=timedelta(days=365),
+    )
+
+    db.session.add_all([BWV, EB, tibber_tariff])
+
+    """
+        Saving TimeBeliefs to the DB
+    """
+
+    # Adding EB, BWV and Tibber Tarriff beliefs to the DB
+    for sensor, source_name, value in [
+        (BWV, "Belastingdienst", 0.21),
+        (EB, "Belastingdienst", 0.12599),
+        (tibber_tariff, "Tibber", 0.018),
+    ]:
+        belief = TimedBelief(
+            sensor=sensor,
+            source=DataSource(source_name),
+            event_value=value,
+            event_start=datetime(2023, 1, 1, tzinfo=utc),
+            belief_time=datetime(2023, 1, 1, tzinfo=utc),
+        )
+
+        db.session.add(belief)
+
+    # DA Prices
+    entsoe = DataSource("ENTSOE")
+    da_prices = Sensor(
+        "DA prices",
+        generic_asset=electricity_price,
+        event_resolution=timedelta(hours=1),
+    )
+    db.session.add(da_prices)
+    da_prices_beliefs = beliefs_from_timeseries(index, entsoe_prices, da_prices, entsoe)
+    db.session.add_all(da_prices_beliefs)
+
+    return
+
+
+def test_tibber_reporter(tibber_test_data):
+    """
+    This test checks if the calculation of the energy prices gets close enough to the ones
+    displayed in Tibber's App.
+    """
+
+    tibber_reporter = TibberReporter(
+        datetime(2023, 4, 13, tzinfo=utc), datetime(2023, 4, 14, tzinfo=utc)
+    )
+
+    result = tibber_reporter.compute()
+
+    # checking that we've get a result for 24 hours
+    assert len(result) == 24
+
+    tibber_app_price_df = (
+        pd.DataFrame(tibber_app_price, index=index, columns=["event_value"]) / 100
+    )
+
+    # checking that (EPEX+EB + Tibber Tariff)*(1+BWV) = Tibber App Price
+    assert (
+        abs(result - tibber_app_price_df).mean().iloc[0] < 0.01
+    )  # difference of less than 1 cent / kWh
```

Comment:
FYI we're still trying to get an explanation for this.

Reply:
Got it! Please add two more transformations in the calculation of the Tibber prices from DA prices: round to 2 decimals, then round to 1 decimal. The prices should then be an exact match.

Reply:
Thanks! It's matching now to the 0.1 cent / kWh (which is close enough).

Reply:
They should be an exact match. I was assuming the rounding was done after converting to the units of the Tibber prices.